google/cloud/dataproc_spark_connect/session.py

```diff
@@ -488,6 +529,29 @@ def _repr_html_(self) -> str:
         </div>
         """
 
+    def _display_operation_link(self, operation_id: str):
+        assert all(
+            [
+                operation_id is not None,
+                self._region is not None,
+                self._active_s8s_session_id is not None,
+                self._project_id is not None,
+            ]
+        )
+
+        url = (
+            f"https://console.cloud.google.com/dataproc/interactive/{self._region}/"
+            f"{self._active_s8s_session_id}/sparkApplications/application/sql;"
+            f"associatedSqlOperationId={operation_id}?project={self._project_id}"
+        )
+
+        html_element = f"""
+                <div>
+                    <p><a href="{url}">Spark UI</a> (Operation: {operation_id})</p>
+                </div>
+                """
+        display(HTML(html_element))
```

Comment:
When run outside of an IPython interactive environment, this will print the HTML contents to stdout.

In that scenario, someone is most likely importing this library into their custom Python application, and in that case they absolutely will not want us to be polluting their stdout with these HTML elements.

I'd suggest instead following the pattern [here](https://github.com/ipython/ipython/blob/d47970f7097668ec89a47120e14625d44b617d68/IPython/core/display_functions.py#L243), but modifying it to not print anything.

e.g.

```py
    try:
        from IPython.core.interactiveshell import InteractiveShell
    except ImportError:
        return

    if not InteractiveShell.initialized():
        return
```